src/shared/crypto/bitcoin/bitcoin.utils.ts

```diff
@@ -37,23 +40,34 @@ export function decodeBitcoinTx(tx: string) {
   return btc.RawTx.decode(hexToBytes(tx));
 }
 
-function getAddressFromWshOutScript(script: Uint8Array) {
-  return btc.programToWitness(0, script.slice(2));
+const concat = P.concatBytes;
+
+function formatKey(hashed: Uint8Array, prefix: number[]): string {
+  return btc.base58check.encode(concat(Uint8Array.from(prefix), hashed));
+}
+
+function getAddressFromWshOutScript(script: Uint8Array, network: BitcoinNetwork) {
+  return btc.programToWitness(0, script.slice(2), network);
 }
 
-function getAddressFromWpkhOutScript(script: Uint8Array) {
-  return btc.programToWitness(0, script.slice(2));
+function getAddressFromWpkhOutScript(script: Uint8Array, network: BitcoinNetwork) {
+  return btc.programToWitness(0, script.slice(2), network);
 }
 
-function getAddressFromTrOutScript(script: Uint8Array) {
-  return btc.programToWitness(1, script.slice(2));
+function getAddressFromTrOutScript(script: Uint8Array, network: BitcoinNetwork) {
+  return btc.programToWitness(1, script.slice(2), network);
 }
 
-export function getAddressFromOutScript(script: Uint8Array) {
+export function getAddressFromOutScript(script: Uint8Array, network: BitcoinNetworkModes) {
   const outScript = btc.OutScript.decode(script);
+  // This appears to be undefined at times?
+  const bitcoinNetwork = getBtcSignerLibNetworkConfigByMode(network);
```

Comment:
Why is this? We have a map value for all the `BitcoinNetworkModes`?

Reply:
I couldn't track this down but def logged undefined and errored unless I used `?.` which now has a lint warning bc it shouldn't be undefined.